docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,422 @@
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: "eslint:recommended",
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+ESLint comes with two predefined configs:
+
+* `eslint:recommended`: the rules recommended by ESLint
+* `eslint:all`: all rules shipped with ESLint
+
+You can also use the `extends` property to extend a custom config. Custom configs can either be paths to local config files or npm package names:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: ["eslint:recommended", "./custom-config.js", "eslint-config-my-config"],
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+#### eslint.config.js
+
+Predefined configs are imported from separate modules into `eslint.config.js` files.
+
+The `recommended` and `all` rules configs are located in the [`@eslint/js`](https://www.npmjs.com/package/@eslint/js) package. You must import this package to use these configs:
+
+```shell
+npm install @eslint/js -D
+```
+
+You can add each of these configs to the exported array or expose specific rules from them.
+
+You must import the modules for local config files and npm package configs with `eslint.config.js`:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customConfig from "./custom-config.js";
+import myConfig from "eslint-config-my-config";
+
+export default [
+    js.configs.recommended,
+    customConfig,
+    myConfig,
+    {
+        rules: {
+            semi: ["warn", "always"]
+        },
+        // ...other config
+    }
+];
+```
+
+Note that because you are just importing JavaScript modules, you can mutate the config objects before exporting them. For example, you might want to have a certain config object only apply to your test files:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customTestConfig from "./custom-test-config.js";
+
+export default [
+    js.configs.recommended,
+    {
+        ...customTestConfig,
+        files: ["**/*.test.js"],
+    },
+];
+```
+
+### Ignoring Files
+
+#### .eslintrc
+
+To make ESLint ignore files with `.eslintrc`, you can create a separate `.eslintignore` file in the root of your project. The `.eslintignore` file uses the same glob pattern syntax as `.gitignore` files.
+
+```shell
+# .eslintignore
+temp.js
+.config/*
+# ...other ignored files
+```
+
+Alternatively, you can use an `ignorePatterns` property in your `.eslintrc` file:
+
+```javascript
+// .eslintrc.js
+{
+    // ...other config
+    ignorePatterns: ["temp.js", ".config/*"],
+}
+```
+
+#### eslint.config.js
+
+To ignore files with `eslint.config.js`, you can use the `ignores` property in a config object:
+
+```javascript
+export default [
+    {
+        // ...other config
+        ignores: ["temp.js", ".config/*"]
+    }
+];
+```
+
+Also, dotfiles (e.g. `.dotfile.js`) are no longer ignored by default with `eslint.config.js`.
```

Comment:
re the dotfiles: in a later tsc meeting, we've decided to keep it as-is:
https://github.com/eslint/tsc-meetings/pull/378/files#diff-86bfb8d4387c1278423e2710f8108a542396d2bc131cce91479a7b3fbfec23e5R27

Reply:
> re the dotfiles: in a later tsc meeting, we've decided to keep it as-is: https://github.com/eslint/tsc-meetings/pull/378/files#diff-86bfb8d4387c1278423e2710f8108a542396d2bc131cce91479a7b3fbfec23e5R27

I think that resolution is about glob patterns, so for example `"ignores": "src/**"` shall not ignore files whose name starts with a dot. If `ignores` pattern are not specified, by default, dot-files are linted as well.

Reply:
i'm not sure if i fully understand but will refactor and y'all can comment on if it needs further refinement.